scripts/envhook.py

```diff
@@ -28,9 +28,16 @@ def main(argv):
             link_dir = os.path.abspath(os.path.join(os.path.dirname(site.__file__), 'site-packages'))
     else:
         raise RuntimeError('Need to be run from within a virtualenv')
-    link = os.path.join(link_dir, 'sitecustomize.py')
     dst = os.path.abspath(__file__)
-    dst = os.path.relpath(dst, link_dir)
+    try:
+        import sitecustomize
+    except ImportError:
+        pass
+    else:
+        if os.path.realpath(sitecustomize.__file__) != dst:
```

Comment:
`realpath` calls `abspath` on Posix and Darwin, the two platforms we care about. I think I can easily create a situation in which `realpath(p1) == realpath(p2) and realpath(p1) != abspath(p2)`. Just make one of the directories in `p2` a symlink. Maybe I'm not getting it but my default would be to only ever compare realpath's. I think my original code was already broken in this way. So what I am proposing is to switch out every use of abspath and replace it with realpath.

Reply:
I see. Ok I'll think this through and make the change.